From cb565ad30abaa6d73d61c81f49bbec5f86042c43 Mon Sep 17 00:00:00 2001
From: Aniket Limaye <a-limaye@ti.com>
Date: Mon, 1 Jun 2026 19:34:28 +0530
Subject: [PATCH] WIP TDA54: board: ti: tda54.env: Apply overlay if requested

If name_overlays is set, apply the overlay from dtboaddr
This only works for a single dtbo loaded to dtboaddr(=0x89000000)

Upstream-Status: Inappropriate [embedded specific]
Signed-off-by: Aniket Limaye <a-limaye@ti.com>
Signed-off-by: Shreyash Sinha <s-sinha@ti.com>
---
 board/ti/tda54/tda54.env | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/board/ti/tda54/tda54.env b/board/ti/tda54/tda54.env
index 7b37057f..6e64fc7a 100644
--- a/board/ti/tda54/tda54.env
+++ b/board/ti/tda54/tda54.env
@@ -16,7 +16,11 @@ xen_boot=0
 xen_addr=0x89000000
 xen_dtboaddr=0x87000000
 run_xen_kern=fdt addr ${fdtaddr}; fdt resize 65536; fdt apply ${xen_dtboaddr}; booti ${xen_addr} - ${fdtaddr}
-run_kern=if test ${xen_boot} = 1; then run run_xen_kern; else booti ${loadaddr} ${rd_spec} ${fdtaddr}; fi
+
+name_overlays=k3-tda54-vdk-vision-apps.dtbo
+apply_overlay=if test "x${name_overlays}" != "x"; then fdt addr ${fdtaddr}; fdt resize 0x100000; fdt apply ${dtboaddr}; fi
+
+run_kern=if test ${xen_boot} = 1; then run run_xen_kern; else run apply_overlay; booti ${loadaddr} ${rd_spec} ${fdtaddr}; fi
 
 boot_targets=mmc1 mmc0 usb pxe dhcp
 boot=mmc
-- 
2.34.1

